src/main/java/command/DeleteCommand.java

```diff
@@ -0,0 +1,46 @@
+package command;
+
+import dukeexeption.InvalidArgumentException;
+import storage.TaskList;
+
+/**
+ * Command component that executes a delete command.
+ */
+public class DeleteCommand extends Command {
+    private int index;
+
+    /**
+     * Constructor for a delete task command.
+     *
+     * @param index index of task to be deleted
+     */
+    public DeleteCommand(int index) {
+        this.index = index;
+    }
+
+    @Override
+    public String run(TaskList taskList) throws InvalidArgumentException {
+        try {
+            String taskDescription = taskList.showTask(this.index).toString();
+            taskList.deleteTask(this.index);
+            return "Noted. I've removed this task:\n  " + taskDescription + "\nNow you have " + taskList.countTask()
```

Comment:
I like how you abstract commands under the abstract class .Seeing how each run statement return a string , maybe store that string in each command and make a output method , this was you abstract all you're outputs there and in the future if you;re output message changes you would not have to change the code .

Reply:
Yeah considered that too and it would be easier for testing as well.